### PR TITLE
Tests for fee related functions in Zk App Commands

### LINKED ISSUE #12625: Write tests for fee related functions
Write unit tests for functions `fee`,`fee_excess`, `fee_payer_account_update` , `fee_payer_pk` in `zkapp_command.ml`

### PR DESCRIPTION
**Pull request by ejMina226**
Branched off another non-develop branch, so ignore all files other than `fee_related.ml` The other branch will be merged first.

Explain your changes:
* Property based tests for the following functions in `src/lib/mina_base/zkapp_command`:
1. `fee`
2. `fee_payer_pk`
3.  `fee_payer_account_update`
4. `fee_excess`

* Closes #12625

### CODE REVIEW

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by ejMina226** [comment]
!ci-build-me

File: src/lib/mina_base/test/zkapp_command/fee_related.ml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+open Snark_params.Tick
+
+let%test_module "fee_related_tests" =
+  ( module struct
+    let feepayer_body_generator = Zkapp_command.T.Stable.Latest.Wire.gen
+
+    let%test_unit "fee" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Currency.Fee.t] x.fee_payer.body.fee
+            (fee @@ Zkapp_command.T.Stable.Latest.of_wire x) )
```
Comment (psteckler) [inline src/lib/mina_base/test/zkapp_command/fee_related.ml:14]: `Zkapp_command.of_wire`

File: src/lib/mina_base/test/zkapp_command/fee_related.ml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    let%test_unit "fee_payer_account_update" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Account_update.Fee_payer.Stable.V1.t] x.fee_payer
```
Comment (psteckler) [inline src/lib/mina_base/test/zkapp_command/fee_related.ml:18]: `Account_update.Fee_payer.t`

Avoid using versioned modules, they're only for defining types

File: src/lib/mina_base/test/zkapp_command/fee_related.ml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    let%test_unit "fee_payer_pk" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Signature_lib.Public_key.Compressed.Stable.V1.t]
+            x.fee_payer.body.public_key
+            (fee_payer_pk @@ Zkapp_command.T.Stable.Latest.of_wire x) )
+
+    let%test_unit "fee_excess" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: (Token_id.t, Currency.Fee.Signed.t) Fee_excess.Poly.t]
```
Comment (psteckler) [inline src/lib/mina_base/test/zkapp_command/fee_related.ml:29]: I think this is the same type as `Fee_excess.t`

File: src/lib/mina_base/test/zkapp_command/fee_related.ml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+open Core_kernel
+open Mina_base.Zkapp_command
+open Mina_base
+open Snarky_backendless
+open Snark_params.Tick
+
+let%test_module "fee_related_tests" =
+  ( module struct
+    let feepayer_body_generator = Zkapp_command.T.Stable.Latest.Wire.gen
```
[the remark this answers is not in the document]
Comment (ejMina226) [reply src/lib/mina_base/test/zkapp_command/fee_related.ml:9]: There doesn't seem to be a gen method on Zkapp_command.

**Comment by ejMina226** [comment]
!ci-build-me

File: src/lib/mina_base/test/zkapp_command/fee_related.ml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    let%test_unit "fee" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Currency.Fee.t] x.fee_payer.body.fee
+            (fee @@ Zkapp_command.of_wire x) )
```
Comment (Sventimir) [inline src/lib/mina_base/test/zkapp_command/fee_related.ml:14]: `Zkapp_command` is already open. However, given that most functions called from this module are from there, I'd prefer to only open it locally. This arrangement gives the impression that `fee` and `of_wire` functions come from different modules, which is not the case. In short, I suggest to rather write: `Zkapp_command.(fee @@ of_wire x)` and remove the `open` at the top of the module.

**Review by Sventimir — COMMENTED** [review]
I'm not really sure, what these tests are meant to check. Are they checking the accessor functions for `Zkapp_command types`? The generator? The accessors are pretty straight forward and these tests seem to just restate their definitions, which does not seem very useful…

File: src/lib/mina_base/test/zkapp_command/fee_related.ml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ejMina226) [reply src/lib/mina_base/test/zkapp_command/fee_related.ml:9]: In zkapp_command.ml:
`[%%define_locally Stable.Latest.(of_wire, to_wire, gen)]`
doesn't seem to work. It doesn't recognise `gen`

[the remark this answers is not in the document]
Comment (ejMina226) [reply src/lib/mina_base/test/zkapp_command/fee_related.ml]: I had to add in a second statement `[%%define_locally Stable.Latest.Wire.(gen)]` and that worked.

**Comment by ejMina226** [comment]
!ci-build-me

**Comment by ejMina226** [comment]
!ci-build-me

**Review by psteckler — APPROVED** [review]

**Comment by ejMina226** [comment]
!ci-build-me

**Pull request merged**